pyteal/compiler/subroutines.py

```diff
@@ -56,12 +57,38 @@ def findRecursionPoints(
         reentryPoints[subroutine] = set(
             callee
             for callee in subroutineGraph[subroutine]
-            if depthFirstSearch(subroutineGraph, callee, subroutine)
+            if graph_search(subroutineGraph, callee, subroutine)
         )
 
     return reentryPoints
 
 
+def find_recursive_path(
+    subroutine_graph: Dict[SubroutineDefinition, Set[SubroutineDefinition]],
+    subroutine: SubroutineDefinition,
+) -> List[SubroutineDefinition]:
+    visited = set()
+    loop = []
+
+    def dfs(x):
```

Comment:
Why create a new method of searching instead of using `graph_search`?

Reply:
I needed to get the entire path from beginning of the recursion back to itself. I found I had to modify `graph_search()` too much for my comfort level - i.e. the risk of changing its behavior was too high for the amount gained of code re-usage. Hence, the new function.

Reply:
`graph_search` doesn't return the entire path, only the parent-child relationships for subroutines that are recursive. I guess an alternative to constructing the entire path from scratch would be to use the output of  `graph_search` and then search again on it to find the path. However, that didn't seem to be any simpler so I didn't pursue it. Do you have an alternate approach in mind?

Reply:
well, as far as I understand it, Zeph is using `find_recursive_path` function to print a full path of self calling subroutine call-path, which fits into the definition of recursive call.

An alternative solution which fits Jason's preference is to extend `graph_search` to return a full path between start and end. Ignoring DFS/BFS, both functions should be able to find a bath between 2 nodes in finite sized connected graph.

Reply:
My mistake, I didn't see that `dfs` modifies the `loop` variable. I'll resolve

Reply:
Well I guess my suggestion would be that you shouldn't modify a captured variable like this because it's easy to miss; instead, returning a `Union[List, None]` or similar would be easier to understand at a glance. But this is purely a suggestion, and I don't mind if you don't take it

Reply:
I see. I think I'll pass up this improvement since I'm trying to wrap a few other things up and it's not a blocker. Am resolving.